internal/flags/apiflags/flags_test.go

```diff
@@ -0,0 +1,295 @@
+package apiflags
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/spf13/pflag"
+	"gotest.tools/assert"
+)
+
+func TestAddTo(t *testing.T) {
+	testCases := []struct {
+		name     string
+		apiFlags APIFlags
+		validate func(APIFlags, *pflag.FlagSet) error
+	}{
+		{
+			// Populate FlagSet
+			name: "Populate FlagSet",
+			apiFlags: APIFlags{
+				APIVersion:       "app.example.com/v1alpha1",
+				Kind:             "AppService",
+				SkipGeneration:   false,
+				CrdVersion:       "v1",
+				HelmChartRef:     "stable/app",
+				HelmChartVersion: "1.0.0",
+				HelmChartRepo:    "https://charts.mycompany.com/",
+			},
+			validate: func(apiFlags APIFlags, flagSet *pflag.FlagSet) error {
+				val, err := flagSet.GetString("api-version")
+				if err != nil {
+					return err
+				}
+				if apiFlags.APIVersion != val {
+					return fmt.Errorf("APIVersion does not match")
+				}
+
+				val, err = flagSet.GetString("kind")
+				if err != nil {
+					return err
+				}
+				if apiFlags.Kind != val {
+					return fmt.Errorf("Kind does not match")
+				}
+
+				boolVal, err := flagSet.GetBool("skip-generation")
+				if err != nil {
+					return err
+				}
+				if apiFlags.SkipGeneration != boolVal {
+					return fmt.Errorf("SkipGeneration does not match")
+				}
+
+				val, err = flagSet.GetString("crd-version")
+				if err != nil {
+					return err
+				}
+				if apiFlags.CrdVersion != val {
+					return fmt.Errorf("CrdVersion does not match")
+				}
+
+				val, err = flagSet.GetString("helm-chart-repo")
+				if err != nil {
+					return err
+				}
+				if apiFlags.HelmChartRepo != val {
+					return fmt.Errorf("HelmChartRepo does not match")
+				}
+
+				val, err = flagSet.GetString("helm-chart-version")
+				if err != nil {
+					return err
+				}
+				if apiFlags.HelmChartVersion != val {
+					return fmt.Errorf("HelmChartVersion does not match")
+				}
+
+				return nil
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		expFlags := &pflag.FlagSet{}
+		t.Run(tc.name, func(t *testing.T) {
+			tc.apiFlags.AddTo(expFlags)
+			if tc.validate != nil {
+				if err := tc.validate(tc.apiFlags, expFlags); err != nil {
+					t.Fatal("unexpected error validating AddTo", err)
+				}
+			}
+		})
+	}
+}
+
+func TestVerifyCommonFlags(t *testing.T) {
+	testCases := []struct {
+		name         string
+		apiFlags     APIFlags
+		operatorType string
+		expError     string
+	}{
+		{
+			// Valid Go API Flags
+			name: "Valid Go API Flags",
+			apiFlags: APIFlags{
+				APIVersion:     "app.example.com/v1alpha1",
+				Kind:           "AppService",
+				SkipGeneration: false,
+			},
+			operatorType: "go",
+			expError:     "",
+		},
+
+		{
+			// InValid Go API Flags
+			name: "InValid Go Kind",
+			apiFlags: APIFlags{
+				APIVersion: "app.example.com/v1alpha1",
+			},
+			operatorType: "go",
+			expError:     "value of --kind must not have empty value",
```

Comment:
typically I don't eve compare error strings. This will cause brittle tests down the road. In reality we don't usually care what the error says but whether the error occurred or not. If these tests pass, we can do a follow-on PR. I have some other examples of dealing with errors.

Reply:
Here is one example where I cared about the error being returned: https://github.com/openshift/ansible-service-broker/blob/master/pkg/broker/broker_test.go#L214-L220

Reply:
Here is an example of where I do not care about the specific error but only that an error is returned under certain conditions. https://github.com/automationbroker/bundle-lib/blob/master/runtime/run_bundle_test.go#L624-L630.